geode-core/src/main/java/org/apache/geode/cache/wan/GatewaySender.java

```diff
@@ -174,7 +174,32 @@
    */
   int GET_TRANSACTION_EVENTS_FROM_QUEUE_RETRIES =
       Integer.getInteger(GeodeGlossary.GEMFIRE_PREFIX + "get-transaction-events-from-queue-retries",
-          10);
+          2);
+  /**
+   * Milliseconds to wait before retrying to get events for a transaction from the
+   * gateway sender queue when group-transaction-events is true.
+   */
+  int GET_TRANSACTION_EVENTS_FROM_QUEUE_WAIT_TIME_MS =
+      Integer.getInteger(
+          GeodeGlossary.GEMFIRE_PREFIX + "get-transaction-events-from-queue-wait-time-ms",
+          1);
+
+  /**
+   * When group-transaction-events is set to true and the gateway sender is stopped,
+   * there is a possibility that the stopping occurs such that for a transaction,
+   * not all events belonging to it reach the queue. The reason would be that
+   * some reach the queue right before the sender is stopped and the rest do not make
+   * it to the queue because the sender is just stopped.
+   * In order to prevent that the queue contains incomplete transactions
+   * due to the above circumstance, this parameter allows for a grace period
+   * of the number of milliseconds set in it before the gateway sender is
+   * actually stopped, in which only events to complete transactions are put in the queue.
+   * Other events received in this period would be dropped.
```

Comment:
Suggested rewrite for economy of language:
When group-transaction-events is true and the gateway sender is stopped,
addition to the queue of a group of transaction events might be interrupted.
To ensure that the queue does not contain incomplete transactions, this parameter
allows for a grace period, specified in milliseconds, before the gateway sender is actually
stopped, allowing complete transaction event groups to be queued. Any event received
during the grace period that are not part of a transaction event group are dropped.

Reply:
Thanks Dave. Very good suggestion. I made a slight change in the last sentence of your proposal but other than that, I took it as is.